clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/DIVStats.java

```diff
@@ -67,71 +70,57 @@ public DIVStats() {
   }
 
   public long getDuplicateMsg() {
-    return duplicateMsg;
+    return this.duplicateMsg.sum();
   }
 
   public void recordDuplicateMsg() {
-    duplicateMsg += 1;
+    this.duplicateMsg.increment();
   }
 
   public void setDuplicateMsg(long count) {
-    this.duplicateMsg = count;
+    this.duplicateMsg.reset();
+    this.duplicateMsg.add(count);
   }
 
-  public long getMissingMsg() {
+  public synchronized long getMissingMsg() {
```

Comment:
Every other place looks good to me, just curious why we need synchronized in getter method here?

Reply:
All accesses to `long` variables should be synchronized, or else the CPU cache can result in stale data. There is a spot bugs warning for that if we partially synchronize. In the case of `long` and `double`, it is especially important since a write can be applied "halfway" (to 4 out of 8 bytes) so a concurrent read could be getting complete non-sense (in theory).

Reply:
Good to learn this! https://stackoverflow.com/questions/517532/writing-long-and-double-is-not-atomic-in-java

Reply:
This holds true on even 64bit JVMs ?

Reply:
That seems be to a JVM implementation specific per [this doc](https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.7). It's true on some 64 bits JVMs.

Reply:
Good to learn about this. I was also thinking keyword "volatile" but seems like it won't work for compound computation like x++;

Reply:
Volatile will guarantee an atomic read (even for longs and doubles) but is not sufficient for thread safety in ++ operation.